central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -439,6 +441,7 @@ func (ds *datastoreImpl) ResolveListDeployments(ctx context.Context, collection
 	if err != nil {
 		return nil, err
 	}
+	paginated.FillPagination(query, pagination, defaultPageSize)
```

Comment:
Avoid fill pagination at datastore or lower layers instead do it at the service layer.